src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -603,7 +604,7 @@ def get_good_view_with_patch_refinement(self) -> bool:
         """
         self.get_good_view("view_finder")
         for patch_id in ("patch", "patch_0"):
-            if patch_id in self._observation["agent_id_0"].keys():
+            if patch_id in self._observation[AgentID("agent_id_0")].keys():
```

Comment:
We don't need to call `keys()` when checking if a dictionary contains a specific key.
```suggestion
            if patch_id in self._observation[AgentID("agent_id_0")]:
```